packages/core/src/lib/modal/Modal.tsx

```diff
@@ -50,6 +50,10 @@ const Modal: React.FC<ModalProps> = ({ options, network, wallets }) => {
     };
   }, []);
 
+  useEffect(() => {
```

Comment:
_Not sure if the PR is ready for review but I have a some suggestions:_

I think we could just call `resetState()` at the end of the `handleDismissClick()` function.
This would avoid unnecessarily reseting the state in 2 cases, with `useEfect` the state is being reset when the component first loads and when we first open the modal.

And maybe it would be nice to clear the ledgerError  `setLedgerError("");` as soon as we click the "Connect" `handleConnectClick()` .
For example let's say I provided an invalid account id after I click connect and confirm it in ledger then the error will be shown at this moment let say I made  a typo in the account id after I fix it and click the connect button again it would be nice to clear the error until ledger throws the next error if there's any.

Reply:
The problem with having the `resetState()` function just inside the `handleDismissClick()` is that when we don't click dismiss and click connect and everything goes correctly. In this case we did not reset the state. Now if we disconnect and try to connect using ledger again the inputs will have the old state data.

The reason we can't just add `resetState()` inside the `handleConnectClick()` is that the user will not see the errors after we reset the state.

Reply:
Yeah that seems to be the case.